internal/storage/v2/memory/memory.go

```diff
@@ -116,21 +116,38 @@ func (st *Store) GetServices(ctx context.Context) ([]string, error) {
 
 func (st *Store) FindTraces(ctx context.Context, query tracestore.TraceQueryParams) iter.Seq2[[]ptrace.Traces, error] {
 	m := st.getTenant(tenancy.GetTenant(ctx))
-	if query.SearchDepth <= 0 || query.SearchDepth > st.defaultConfig.MaxTraces {
+	traceAndIds, err := m.findTraceAndIds(query)
+	if err != nil {
 		return func(yield func([]ptrace.Traces, error) bool) {
-			yield(nil, errInvalidSearchDepth)
+			yield(nil, err)
 		}
 	}
-	traces := m.findTraces(query)
 	return func(yield func([]ptrace.Traces, error) bool) {
-		for _, trace := range traces {
-			if !yield([]ptrace.Traces{trace}, nil) {
+		for _, trace := range traceAndIds {
+			if !yield([]ptrace.Traces{trace.trace}, nil) {
```

Comment:
avoid copying trace by value twice
```suggestion
		for i := range traceAndIds {
			if !yield([]ptrace.Traces{traceAndIds[i].trace}, nil) {
```